### PR TITLE
Memory overflows in cli mode

### PR DESCRIPTION
**Pull request by hughcube**
Memory overflows in cli mode

| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | yes
| Tests pass?   | yes
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

### CODE REVIEW

File: framework/caching/ArrayCache.php [hunk 1]
```diff
@@ -84,6 +104,26 @@ protected function deleteValue($key)
...
+     * Removes the expired data values.
+     * @param bool $force whether to enforce the garbage collection regardless of [[gcProbability]].
+     * Defaults to false, meaning the actual deletion happens with the probability as specified by [[gcProbability]].
+     */
+    public function gc($force = false)
+    {
+        if ($force || mt_rand(0, 1000000) < $this->gcProbability){
+            foreach($this->_cache as $key => $item){
+                if (!isset($item[0], $item[1]) || (0 !== $item[0] && $item[1] < microtime(true))){
+                    $this->deleteValue($key);
```
Comment (rob006) [inline framework/caching/ArrayCache.php]: You have potential infinite recursion here: `gc()` calls `deleteValue()` which calls `gc()`.

**Review by rob006 — DISMISSED** [review]

File: framework/caching/ArrayCache.php (hunk 1 above)
Comment (hughcube) [reply framework/caching/ArrayCache.php]: Has been fixed

**Comment by rob006** [comment]
Honestly, I'm not sure about the whole concept. AFAIK `ArrayCache` was designed as super fast dummy cache, and adding GC (especially enabled by default) in 99.99% cases will give you nothing except additional overhead. Especially that it is triggered more often than in other cache backends.

**Comment by samdark** [comment]
@hughcube are you running daemons?

**Comment by hughcube** [comment]
> are you running daemons?

Yes, otherwise there won't be this demand

**Comment by cebe** [comment]
Could be disabled by default. Also GC value depends a lot on the number of calls to the cache, it could have a time limit also, which would prevent a GC to run more than x times per time frame.

**Comment by samdark** [comment]
That seems to be out of scope of the common framework usage. Thanks for contribution, @hughcube but won't be merged.

**Pull request closed without merging**